bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/AbstractLinkRegistry.java

```diff
@@ -25,16 +31,89 @@
  *
  * @author Dennis Nobel - Initial contribution
  *
- * @param <L>
- *            Concrete type of the abstract link
+ * @param <L> Concrete type of the abstract link
  */
 public abstract class AbstractLinkRegistry<L extends AbstractLink, P extends Provider<L>>
         extends AbstractRegistry<L, String, P> {
 
+    private final ReentrantReadWriteLock toLinkLock = new ReentrantReadWriteLock();
+    private final Map<String, Set<L>> itemNameToLink = new HashMap<>();
```

Comment:
It looks like you could use a itemNameToUid here because this set is primarily used to get the uid.

Reply:
The member is defined as a map that provides a set of links for an item name.
If we currently use the links to get the linked UID it does not matter.

Reply:
The current logic works also for duplicates as the set stores the link and not the UID.
So on removal the removed element (the link) is removed from the set.

Reply:
Ah I see what you mean. If 2 links have the same uid linked that would be hard to remove with a lot of checking code.